dask/dataframe/shuffle.py

```diff
@@ -776,6 +773,49 @@ def set_index_post_series(df, index_name, drop, column_dtype):
     return df2
 
 
+def drop_overlap(df, index):
+    return df.drop(index) if index in df.index else df
+
+
+def get_overlap(df, index):
+    return df.loc[[index]] if index in df.index else None
+
+
+def fix_overlap(ddf, overlap):
+    """ Ensures that the upper bound on each partition of ddf is exclusive """
+    name = "fix-overlap-" + tokenize(ddf, overlap)
+    n = len(ddf.divisions) - 1
+    dsk = {(name, i): (ddf._name, i) for i in range(n)}
+
+    for i in overlap:
+        frames = [(get_overlap, (ddf._name, i - 1), ddf.divisions[i]), (ddf._name, i)]
+        dsk[(name, i)] = (pd.concat, frames)
+        dsk[(name, i - 1)] = (drop_overlap, dsk[(name, i - 1)], ddf.divisions[i])
+
+    graph = HighLevelGraph.from_collections(name, dsk, dependencies=[ddf])
+    return new_dd_object(graph, name, ddf._meta, ddf.divisions)
+
+
+def compute_divisions(df, **kwargs):
```

Comment:
Since this both computes and sets divisions (rather than just computes them), perhaps rename to `with_computed_divisions` or `compute_and_set_divisions`.